src/osd/OSDMap.h

```diff
@@ -1164,7 +1164,7 @@ class OSDMap {
    * This does not apply temp overrides and should not be used
    * by anybody for data mapping purposes. Specify both pointers.
    */
-  void pg_to_raw_up(pg_t pg, vector<int> *up, int *primary) const;
+  void pg_to_raw_up(pg_t pg, vector<int> *up, int *primary, bool to_up) const;
```

Comment:
Wouldn't it be simpler to make `to_up = true` ?

Reply:
All right.I will modify it according to what you said.Can you help me check if there is any problem with this modification

Reply:
I have modified it according to your suggestion

Reply:
Seems ```maybe_remove_pg_upmaps``` will only remove an upmap item if it causes collided mappings, which means we basically shouldn't exclude __down__ OSDs here.

Can you promote the OSD subsys's log level to 10 and post a more verbose log onto the tracker?

Reply:
I submitted the log to tracker.Please have a look first.